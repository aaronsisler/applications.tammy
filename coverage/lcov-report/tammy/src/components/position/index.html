<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for tammy/src/components/position</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> tammy/src/components/position
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>35/42</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.86% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>13/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">73.08% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>19/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.59% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>35/37</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="PositionContainer.js"><a href="PositionContainer.js.html">PositionContainer.js</a></td>
	<td data-value="85.71" class="pic high"><div class="chart"><div class="cover-fill" style="width: 85%;"></div><div class="cover-empty" style="width:15%;"></div></div></td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="PositionDetails.js"><a href="PositionDetails.js.html">PositionDetails.js</a></td>
	<td data-value="83.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 83%;"></div><div class="cover-empty" style="width:17%;"></div></div></td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="PositionDetailsContent.js"><a href="PositionDetailsContent.js.html">PositionDetailsContent.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="PositionList.js"><a href="PositionList.js.html">PositionList.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file medium" data-value="PositionListFilter.js"><a href="PositionListFilter.js.html">PositionListFilter.js</a></td>
	<td data-value="62.5" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 62%;"></div><div class="cover-empty" style="width:38%;"></div></div></td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

<tr>
	<td class="file medium" data-value="PositionListItem.js"><a href="PositionListItem.js.html">PositionListItem.js</a></td>
	<td data-value="71.43" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

<tr>
	<td class="file high" data-value="PositionsPage.js"><a href="PositionsPage.js.html">PositionsPage.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 16 2018 21:02:55 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
<script src="..\..\..\..\block-navigation.js"></script>
</body>
</html>
