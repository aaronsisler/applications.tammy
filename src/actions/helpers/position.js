export const clearPosition = () => ({
    type: 'CLEAR_POSITION'
});

export const setPosition = (position) => ({
    type: 'SET_POSITION',
    position
});
