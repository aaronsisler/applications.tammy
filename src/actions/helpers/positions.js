export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
