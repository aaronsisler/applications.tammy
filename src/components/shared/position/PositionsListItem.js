import React from 'react';
import PropTypes from 'prop-types';
import history from 'Tools/history';

export class PositionsListItem extends React.Component {
    constructor(props) {
        super(props);
    }

    handleNavigate = () => {
        const { linkRoute, positionId } = this.props;
        history.push(`/${linkRoute}/${positionId}`);
    }

    render() {
        return (
            <div className="positions_list_item" onClick={this.handleNavigate}>
                <div className="positions_list_item__content">
                    <div className="positions_list_item__title">
                        {this.props.title}
                    </div>
                    <div className="positions_list_item__job_id">
                        Job Id: {this.props.jobId}
                    </div>
                </div>
                <div className="positions_list_item__location">
                    {this.props.location}
                </div>
            </div>
        );
    }
}

export default PositionsListItem;

PositionsListItem.propTypes = {
    jobId: PropTypes.string.isRequired,
    linkRoute: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    positionId: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};
