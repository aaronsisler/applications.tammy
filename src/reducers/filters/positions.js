const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
