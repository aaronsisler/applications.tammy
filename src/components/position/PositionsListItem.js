import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { startSetPosition } from 'Actions/position';

export class PositionsListItem extends React.Component {
    constructor(props) {
        super(props);
    }

    handleSetPosition = () => {
        this.props.startSetPosition(this.props.positionId);
    }

    render() {
        return (
            <div className="positions_list_item" onClick={this.handleSetPosition}>
                <div className="positions_list_item__content">
                    <div className="positions_list_item__title">
                        {this.props.title}
                    </div>
                    <div className="positions_list_item__job_id">
                        Job Id: {this.props.jobId}
                    </div>
                </div>
                <div className="positions_list_item__location">
                    {this.props.location}
                </div>
            </div>
        );
    }
}

/* istanbul ignore next */
const mapDispatchToProps = (dispatch) => ({
    startSetPosition: (positionId) => dispatch(startSetPosition(positionId))
});

export default connect(undefined, mapDispatchToProps)(PositionsListItem);

PositionsListItem.propTypes = {
    jobId: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    positionId: PropTypes.string.isRequired,
    startSetPosition: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
};
