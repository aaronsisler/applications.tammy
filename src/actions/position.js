import { setPosition, clearPosition } from 'Actions/helpers/position';

export const startClearPosition = () => (dispatch) => dispatch(clearPosition());

export const startSetPosition = (positionId) => (dispatch, getState) => {
    const { positions } = getState();
    const positionMatch = positions.find((position) => position.positionId == positionId);
    return dispatch(setPosition(positionMatch));
}
