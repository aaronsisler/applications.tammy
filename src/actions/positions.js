import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
