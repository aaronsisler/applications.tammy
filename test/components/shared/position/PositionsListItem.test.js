import React from 'react';
import { shallow } from 'enzyme';
import { PositionsListItem, mapDispatchToProps } from 'Shared/position/PositionsListItem';
import positions from '../../../fixtures/positions';

import * as positionActions from 'Actions/position';

describe('PositionsListItem', () => {
    const startSetPosition = jest.fn();
    const [position] = positions;
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <PositionsListItem
                {...position}
                startSetPosition={startSetPosition}
            />);
    });

    it('should render PositionsListItem correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call startSetPosition prop when clicked', () => {
        wrapper.find('.positions_list_item').simulate('click');

        expect(startSetPosition).toHaveBeenCalled();
    });

    describe('mapDispatchToProps', () => {
        it('should call startSetPosition with positionId', () => {
            const startSetPositionMock = jest.spyOn(positionActions, 'startSetPosition');
            const dispatch = jest.fn();
            const { positionId } = position;

            mapDispatchToProps(dispatch, { positionId }).startSetPosition();

            expect(startSetPositionMock).toHaveBeenLastCalledWith(positionId);
        });
    });
});
