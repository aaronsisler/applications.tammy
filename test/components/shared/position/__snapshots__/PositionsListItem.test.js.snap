// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionsListItem should render PositionsListItem correctly 1`] = `
<div
  className="positions_list_item"
  onClick={[Function]}
>
  <div
    className="positions_list_item__content"
  >
    <div
      className="positions_list_item__title"
    >
      first title
    </div>
    <div
      className="positions_list_item__job_id"
    >
      Job Id: 
      first job id
    </div>
  </div>
  <div
    className="positions_list_item__location"
  >
    first location
  </div>
</div>
`;
